### PR TITLE
chore(operations): Add nightly GH action workflow

### PR DESCRIPTION
**Pull request by binarylogic**

### CODE REVIEW

**Comment by binarylogic** [comment]
🎉 - see @lukesteensen, the celebration comes after and everything works.

**Comment by binarylogic** [comment]
Before we merge I want to test that Docker and S3 artifacts are uploaded properly.

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
+name: nightly
+on:
+  push:
+    branches:
+      - master
+  pull_request: {}
+
+# on:
```
Comment (Hoverbear) [inline .github/workflows/nightly.yml:8]: I think we need to uncomment this for it to work. :)

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+          path: "./target/artifacts/vector-x86_64-apple-darwin.tar.gz"
+
+  build-x86_64-pc-windows-msvc-archive-and-msi-package:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: "Download Perl"
+        shell: bash
+        run: |
+          curl -sSf http://strawberryperl.com/download/$VERSION/strawberry-perl-$VERSION-64bit.msi > perl-installer.msi
```
Comment (Hoverbear) [inline .github/workflows/nightly.yml:87]: Ahh you are very through, I would have used `scoop` here but that would mean we have less control over versions.

Comment (binarylogic) [reply .github/workflows/nightly.yml:87]: This is all @a-rodin

**Comment by Hoverbear** [comment]
This looks great!

**Pull request merged**